src/Engine/Manager/ScriptManager.cpp

```diff
@@ -563,7 +563,42 @@ void ScriptManager::FillPropertyMap(Script* script) {
     } else {
 
         CreateInstance(script);
-        script->FillPropertyMap();
+
+        int propertyCount = script->instance->GetPropertyCount();
+
+        for (int n = 0; n < propertyCount; n++) {
+
+            std::string name = script->instance->GetPropertyName(n);
+            int typeId = script->instance->GetPropertyTypeId(n);
+            void *varPointer = script->instance->GetAddressOfProperty(n);
+
+            if (script->isInPropertyMap(name, typeId))
+                continue;
+
+            if (typeId == asTYPEID_INT32) {
+                int size = sizeof(int);
+                script->AddToPropertyMap(name, typeId, size, varPointer);
+            }
+            else if (typeId == asTYPEID_FLOAT) {
+                int size = sizeof(float);
+                script->AddToPropertyMap(name, typeId, size, varPointer);
+            }
+            else if (typeId == engine->GetTypeIdByDecl("Entity@")) {
+                int size = sizeof(unsigned int);
+                
+                const std::vector<Entity*> entities = Hymn().world.GetEntities();
+                
+                unsigned int* GUID = new unsigned int();
```

Comment:
Instead of making a heap allocation this could be a regular stack variable. The data is ultimately copied after being passed to `AddToPropertyMap` so we don't specifically need the heap. Remember to remove the delete statement.